### PR TITLE
[fix](decimal) Fix wrong result produced by decimal128 multiply

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Now:
image

After fixing:
image

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

File: be/src/vec/common/arithmetic_overflow.h [hunk 1]
```diff
@@ -123,7 +123,7 @@ inline bool mul_overflow(__int128 x, __int128 y, __int128& res) {
 
     unsigned __int128 a = (x > 0) ? x : -x;
     unsigned __int128 b = (y > 0) ? y : -y;
-    return (a * b) / b != a;
+    return ((a * b) / b != a) || (x > 0 && y > 0 && res < 0) || (x < 0 && y < 0 && res < 0);
```
Comment (yiguolei) [inline be/src/vec/common/arithmetic_overflow.h:126]: add some regression test

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by Gabriel39** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Pull request merged**